pkg/importer/dataStream_test.go

```diff
@@ -25,6 +46,78 @@ func createDataStream(ep, accKey, secKey string) *dataStream {
 	return ds
 }
 
+func createDataStreamMulti(testfile, ep string) *dataStream {
+	// set our directory for images
+	imageDir, _ := filepath.Abs("../../test/images")
+	urlPath := filepath.Join(ep, testfile)
+	testFilePath := filepath.Join(imageDir, testfile)
+
+	dsurl, _ := ParseEndpoint(urlPath)
+
+	ds := &dataStream{
+		Url:         dsurl,
+		buf:         make([]byte, image.MaxExpectedHdrSize),
+		accessKeyId: "",
+		secretKey:   "",
+	}
+
+	ds.constructReaders()
+
+	buf := bytes.NewBuffer(nil)
+	f, _ := os.Open(testFilePath)
+	io.Copy(buf, f)
+	f.Close()
+	testBytes, _ := ioutil.ReadAll(buf)
+
+	//append readers
+	xzreader, _ := xz.NewReader(bytes.NewReader(testBytes), 0)
```

Comment:
I don't think you need `bytes.NewReader` here. What about?
```
buf := bytes.NewBuffer(nil)
f, _ := os.Open(testFilePath)
// check err
defer f.Close()
xzreader, _ := xz.NewReader(buf, 0)   # handle err
tarreader := tar.NewReader(xzreader)
ds.appendReader(RdrXz, xzreader)
ds.appendReader(RdrTar, tarreader)
```
But I still don't know why this is needed when `constructReaders` is called above.